grr/server/grr_response_server/gui/api_plugins/metadata_test.py

```diff
@@ -0,0 +1,382 @@
+
+    expected_binarystream = {"type": "string", "format": "binary"}
+    expected["BinaryStream"] = expected_binarystream
+
+    self.assertDictEqual(expected, openapi_primitives)
+
+  def testRepetitiveField(self):
+    # Test the GET route. The repeated field should be a query parameter.
+    params = (self.openapi_desc_dict.get("paths").get("/metadata_test/method4")
+              .get("get").get("parameters"))
+    self.assertEqual(len(params), 1)
+
+    param = params[0]
+    self.assertEqual(param["in"], "query")
+    schema = param["schema"]
+    self.assertEqual(schema["type"], "array")
+    self.assertEqual(schema["items"]["type"], "integer")
+    self.assertEqual(schema["items"]["format"], "int64")
+
+    # Test the POST route. The repeated field should be in the request body.
+    req_body = (self.openapi_desc_dict.get("paths")
+                .get("/metadata_test/method4").get("post").get("requestBody"))
+    req_schema = req_body["content"]["application/json"]["schema"]
+    self.assertEqual(req_schema["type"], "object")
+    schema = req_schema["properties"]["field_repeated"]
+    self.assertEqual(schema["type"], "array")
+    self.assertEqual(schema["items"]["type"], "integer")
+    self.assertEqual(schema["items"]["format"], "int64")
+
+  def testEnumField(self):
+    # Test the GET route. The enum field should be a query parameter.
+    params = (self.openapi_desc_dict.get("paths").get("/metadata_test/method5")
+              .get("get").get("parameters"))
+    self.assertEqual(len(params), 1)
+
+    param = params[0]
+    self.assertEqual(param["in"], "query")
+    ref_get = param["schema"]["$ref"]
+    ref_nodes = ref_get.split("/")
+    ref_nodes = ref_nodes[1:]
+    schema = self.openapi_desc_dict
+    for node in ref_nodes:
+      schema = schema[node]
+
+    # Test the OpenAPI component agains its protobuf counterpart.
+    self.assertEqual(schema["type"], "integer")
+    self.assertEqual(schema["format"], "int32")
+    enum_values_openapi = set(schema["enum"])
+
+    message_d = MetadataEnumFieldMessage.protobuf.DESCRIPTOR
+    self.assertEqual(len(message_d.enum_types), 1)
+    enum_d = message_d.enum_types[0]
+    enum_values_ds = enum_d.values
+    enum_values_protobuf = set(
+      [enum_value_d.number for enum_value_d in enum_values_ds])
+    self.assertSetEqual(enum_values_openapi, enum_values_protobuf)
+
+    # Test the POST route. The enum field should be in the request body.
+    # Just check that the structure is correct and that the same component
+    # is referenced as above.
+    req_body = (self.openapi_desc_dict.get("paths")
+                .get("/metadata_test/method5").get("post").get("requestBody"))
+    req_schema = req_body["content"]["application/json"]["schema"]
+    self.assertEqual(req_schema["type"], "object")
+    ref_post = req_schema["properties"]["field_enum"]["$ref"]
+    self.assertEqual(ref_post, ref_get)
+
+  def testOpenApiTypeReferences(self):
```

Comment:
Same comment about the test name. Should we use something like `testReferencesInCompositeTypesAreCorrectlyProcessed` or something along these lines?

Reply:
I totally agree and made the change. Thank you!